File: eco-core/core-nms/v1_16_R3/src/main/java/com/willfp/eco/proxy/v1_16_R3/FastItemStackFactory.java
package com.willfp.eco.proxy.v1_16_R3;

import com.willfp.eco.core.fast.FastItemStack;
import com.willfp.eco.proxy.FastItemStackFactoryProxy;
import com.willfp.eco.proxy.v1_16_R3.fast.EcoFastItemStack;
import org.bukkit.inventory.ItemStack;
import org.jetbrains.annotations.NotNull;

public final class FastItemStackFactory implements FastItemStackFactoryProxy {
    @Override
    public FastItemStack create(@NotNull final ItemStack itemStack) {
        return new EcoFastItemStack(itemStack);
    }
}

File: eco-core/core-nms/v1_16_R3/src/main/kotlin/FastItemStackFactory.kt

import com.willfp.eco.core.fast.FastItemStack
import com.willfp.eco.proxy.FastItemStackFactoryProxy
import com.willfp.eco.proxy.v1_16_R3.NMSFastItemStack
import org.bukkit.inventory.ItemStack

class FastItemStackFactory : FastItemStackFactoryProxy {
    override fun create(itemStack: ItemStack): FastItemStack {
        return NMSFastItemStack(itemStack)
    }
}